src/entt/entity/mixin.hpp

```diff
@@ -130,7 +130,12 @@ class basic_sigh_mixin final: public Type {
     }
 
     void bind_any(any value) noexcept final {
-        owner = internal::any_to_owner<registry_type>(value);
+        if(auto *registry = any_cast<owner_type>(&value)) {
```

Comment:
I'd rather change the code of `any_to_owner` than duplicate the logic in two places.
Also, I agree that `reinterpret_cast` is a no-go. Probably a `static_cast` is more than enough and slightly safer.